label_studio/users/serializers.py

```diff
@@ -18,7 +19,26 @@ def get_avatar(self, user):
         return user.avatar_url
 
     def get_initials(self, user):
-        return user.get_initials()
+        if flag_set('fflag_feat_all_optic_114_soft_delete_for_churned_employees', user=user):
+            return user.get_initials(self._is_deleted(user))
+        else:
+            return user.get_initials()
+
+    def _is_deleted(self, instance):
+        if organization_members := self.context.get('organization_members', None):
+            organization_member_for_user = next(
+                (
+                    organization_member
+                    for organization_member in organization_members
+                    if organization_member.user_id == instance.id
+                ),
+                None,
+            )
```

Comment:
So you have an `organization_members` iterable that you're filtering to find the `organization_member` object that for  user `instance` ?

I think I understand it, but I had to stare it for quite a while to get it. A comment to explain it might be helpful? Even better if it was it's own function or method so that you don't keep iterating through `organization_members` even after you get a `organization_member.user_id == instance.id` match.

Reply:
Yes I'll revisit and either add a comment or try to modify the code so its more clarifying (we can do this with a list with a break though I'm not sure if I prefer that).  This code is using a generator so we should only iterate once (via `next`)

Reply:
Ah cool that makes sense. It'll quit the iteration when the match is found :+1: